pydantic_core/core_schema.py

```diff
@@ -1210,7 +1213,9 @@ def set_schema(
         items_schema: The value must be a set with items that match this schema
         min_length: The value must be a set with at least this many items
         max_length: The value must be a set with at most this many items
-        generator_max_length: The value must be a set with at most this many items
+        generator_max_length: At most this many items will be read from a generator before failing validation
+            This is important because generators can be infinite, and even with a max_length on the set,
+            an infinite generator could run forever producing a finite set of distinct values.
```

Comment:
Technically non-distinct, otherwise I think the set max length check will be hit.

Reply:
my point was even if it runs infinitely it may only produce finitely many that are distinct. But I agree this could be phrased better.. let me think